packages/components/src/Popover/Popover.tsx

```diff
@@ -45,6 +45,26 @@ const isRenderProp = (
   children: ReactNode | PopoverRenderProp
 ): children is PopoverRenderProp => typeof children === 'function'
 
+export const popoverPropKeys = [
```

Comment:
If you move this array down below `PopoverProps` and add this type, it'll be slightly safer, though it could still be missing props. I've asked for help on deriving a type that's an exhaustive array, but it's tricky so this I think is good enough for now.

Also I think `escapeWithReference` should be there. My bad I missed it in my `Menu` restructure, but I could think of a situation where you'd want to use it.
```suggestion
export const popoverPropKeys: Array<keyof PopoverProps> = [
  'escapeWithReference',
```

Reply:
Discussed over chat, but it looks like `escapeWithReference` was deprecated from `Popover`. Keeping it deleted from `MenuProps` seems like the right call.

Did move down the keys array though to make use of the interface + keyof!

Resolved in ef164e1